universe/ShipDesign.cpp

```diff
@@ -650,11 +650,13 @@ HullTypeManager::HullTypeManager() {
         throw;
     }
 
-    TraceLogger() << "Hull Types:";
-    for (const auto& entry : m_hulls) {
-        const HullType* h = entry.second;
-        TraceLogger() << " ... " << h->Name();
-    }
+    TraceLogger() << [&]() {
+            std::string retval("Hull Types:");
```

Comment:
I prefer explicit capture of `[this]` here.

In this location it make no difference.

Implicit capture as a practice throughout the code base, may easily lead to capturing a variable with shorter scope than the lambda.  Given that the code  is written by many hands, there are many differing uses of scope and commonly using unnamed capture will make it harder to debug crashes due to a lambda referencing an out of scope object.